modules/loadbalancer/main.tf

```diff
@@ -38,7 +38,7 @@ locals {
 
   # Recalculate the main input map, taking into account whether the boolean condition is true/false.
   frontend_ips = { for k, v in var.frontend_ips : k => {
-    name                          = try(v.create_public_ip, false) ? azurerm_public_ip.this[k].name : k
+    name                          = k
```

Comment:
refactor(loadbalancer): use map key

The `azurerm_public_ip.this[k].name` is always equal to `k`, hence the expression
becomes much simpler.